analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Constraints/ParameterReassignedConstraint.cs

```diff
@@ -0,0 +1,30 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Constraints;
+
+internal sealed class ParameterReassignedConstraint : SymbolicConstraint
+{
+    public static readonly ParameterReassignedConstraint Instance = new(ConstraintKind.ParameterReassigned);
+
+    public override SymbolicConstraint Opposite => throw new NotSupportedException();
```

Comment:
Other constraints return null in case there is no opposite. We should do the same here as the engine calls `Opposite` without looking at the constraint type in, e.g., negations.
```suggestion
    public override SymbolicConstraint Opposite => null;
```